crates/uv/tests/tool_install.rs

```diff
@@ -2345,6 +2417,165 @@ fn tool_install_malformed() {
         entrypoints = [
             { name = "pybabel", install-path = "[TEMP_DIR]/bin/pybabel" },
         ]
+
+        [tool.options]
+        exclude-newer = "2024-03-25T00:00:00Z"
+        "###);
+    });
+}
+
+/// Test installing, then re-installing with different settings.
+#[test]
+fn tool_install_settings() {
+    let context = TestContext::new("3.12")
+        .with_filtered_counts()
+        .with_filtered_exe_suffix();
+    let tool_dir = context.temp_dir.child("tools");
+    let bin_dir = context.temp_dir.child("bin");
+
+    // Install `black`
+    uv_snapshot!(context.filters(), context.tool_install()
+        .arg("flask>=3")
+        .arg("--resolution=lowest-direct")
+        .env("UV_TOOL_DIR", tool_dir.as_os_str())
+        .env("XDG_BIN_HOME", bin_dir.as_os_str())
+        .env("PATH", bin_dir.as_os_str()), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    warning: `uv tool install` is experimental and may change without warning
+    Resolved [N] packages in [TIME]
+    Prepared [N] packages in [TIME]
+    Installed [N] packages in [TIME]
+     + blinker==1.7.0
+     + click==8.1.7
+     + flask==3.0.0
+     + itsdangerous==2.1.2
+     + jinja2==3.1.3
+     + markupsafe==2.1.5
+     + werkzeug==3.0.1
+    Installed 1 executable: flask
+    "###);
+
+    tool_dir.child("flask").assert(predicate::path::is_dir());
+    tool_dir
+        .child("flask")
+        .child("uv-receipt.toml")
+        .assert(predicate::path::exists());
+
+    let executable = bin_dir.child(format!("flask{}", std::env::consts::EXE_SUFFIX));
+    assert!(executable.exists());
+
+    // On Windows, we can't snapshot an executable file.
+    #[cfg(not(windows))]
+    insta::with_settings!({
+        filters => context.filters(),
+    }, {
+        assert_snapshot!(fs_err::read_to_string(executable).unwrap(), @r###"
+        #![TEMP_DIR]/tools/flask/bin/python
+        # -*- coding: utf-8 -*-
+        import re
+        import sys
+        from flask.cli import main
+        if __name__ == "__main__":
+            sys.argv[0] = re.sub(r"(-script\.pyw|\.exe)?$", "", sys.argv[0])
+            sys.exit(main())
+        "###);
+
+    });
+
+    insta::with_settings!({
+        filters => context.filters(),
+    }, {
+        // We should have a tool receipt
+        assert_snapshot!(fs_err::read_to_string(tool_dir.join("flask").join("uv-receipt.toml")).unwrap(), @r###"
+        [tool]
+        requirements = [{ name = "flask", specifier = ">=3" }]
+        entrypoints = [
+            { name = "flask", install-path = "[TEMP_DIR]/bin/flask" },
+        ]
+
+        [tool.options]
+        resolution = "lowest-direct"
+        exclude-newer = "2024-03-25T00:00:00Z"
+        "###);
+    });
+
+    // Reinstall with `highest`. This is a no-op, since we _do_ have a compatible version installed.
```

Comment:
Hard to say what the right behavior is here. We could also consider discarding the tool whenever settings differ.

Reply:
This is really tough. It seems okay to change after some user feedback. I'm leaning towards consistent `--upgrade` / `--reinstall` semantics where it is required to change the versions. This matches `uv pip install`, yes?

Reply:
Yeah

Reply:
Now that we have `upgrade`, I'm sort of open to making `install` more aggressive about replacing though.

Reply:
That makes sense too.

Reply:
So we could make it replace whenever the settings or versions differ...